homeassistant/helpers/entity.py

```diff
@@ -24,13 +24,20 @@ def generate_entity_id(entity_id_format: str, name: Optional[str],
                        current_ids: Optional[List[str]]=None,
                        hass: Optional[HomeAssistant]=None) -> str:
     """Generate a unique entity ID based on given entity IDs or used IDs."""
-    name = (name or DEVICE_DEFAULT_NAME).lower()
     if current_ids is None:
         if hass is None:
             raise ValueError("Missing required parameter currentids or hass")
 
         current_ids = hass.states.entity_ids()
 
+    return async_generate_entity_id(entity_id_format, name, current_ids)
+
+
+def async_generate_entity_id(entity_id_format: str, name: Optional[str],
```

Comment:
This is a side effect from my quick exploration into making EntityComponent async. Leaving it here for the next PR ;-) Yes, not using run_callback_threadsafe as it is not needed. No async calls are used. Had to split it up because the threaded one is depending on a call to `hass`.